### PR TITLE
[Ready] Prisoner role

### PR DESCRIPTION
**Pull request by RaveRadbury**
## About The Pull Request
I observed someone request to be perma'd round start and then they just chilled in perma the whole shift. Security isn't always going to be up for putting people in perma and players who have been placed in perma tend to go catatonic. This allows players who are interested in having a chill shift doing the stuff in perma and acting like they've done crimes the opportunity to do that.

## Why It's Good For The Game
Crew content. I'm not saying it's mindblowing but if we add it in as a role it will open the door for people to look at perma and what could be done with prisoners.

Prisoners spawn with a prison ID, a jumpsuit, and shoes. They are also currently spawning with a backpack and I'm uncertain about whether or not I want to do anything with that.

This is ready for a testmerge and worked for me locally. 

Thanks to Tralezab for assistance with merge conflicts.
Thanks to EOBGames for the plasmaman prisoner sprites.
Thanks to Quill-Weave for encouragement.

Feedback is of course welcome, and I'd appreciate suggestions re: balancing this role. Maybe they should be roundstart pacified or something.

## Changelog
:cl:
add: Adds prisoner role
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
access.dm file has a thing that generates hud icons for every job + prisoner and erts that ive added
since prisoner is now a job you should remove it from that list

File: code/modules/jobs/job_types/prisoner.dm [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	display_order = JOB_DISPLAY_ORDER_PRISONER
+
+/datum/outfit/job/prisoner
+	name = "prisoner"
+	jobtype = /datum/job/prisoner
+
+	uniform = /obj/item/clothing/under/rank/prisoner
+	shoes = /obj/item/clothing/shoes/sneakers/orange
+	id = /obj/item/card/id/prisoner
+	ears = 0
```
Comment (Fikou) [inline code/modules/jobs/job_types/prisoner.dm]: shouldnt these be null

File: code/modules/jobs/job_types/prisoner.dm [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	total_positions = 3
+	spawn_positions = 3
+	supervisors = "the head of security and warden"
+
+	outfit = /datum/outfit/job/prisoner
+
+	display_order = JOB_DISPLAY_ORDER_PRISONER
+
+/datum/outfit/job/prisoner
+	name = "prisoner"
```
Comment (Fikou) [inline code/modules/jobs/job_types/prisoner.dm]: this should be capitalized

File: code/modules/jobs/job_types/prisoner.dm (hunk 1 above)
Comment (RaveRadbury) [reply code/modules/jobs/job_types/prisoner.dm]: They probably should be. Thanks for taking the time to review.

**Comment by Pugie** [comment]
This gives me horrible barotrauma flashbacks.

**Comment by Shadowflame909** [comment]
Extended Shitcurity fix.

Wish there was a way to make it whitelist only. Or this will turn into assistants 2.0

**Comment by Ziiro** [comment]
I feel like this role should require a number of hours as assistant.

The only other thing I might critique on is I do feel like they should get a PDA. I mean, in theory they can spawn as antag, yea?

**Comment by Shadowflame909** [comment]
Headset Uplink Ziiro

**Comment by Ziiro** [comment]
They don't spawn with headsets. They would default to the.. Pen one I guess?

In either case goddamn that's going to be a fun hard mode.

**Comment by Shadowflame909** [comment]
My only suggestion that doesn't pertain to balance is to have them spawn with a randomly generated crime in there file, and for it to be in there memory as well.

We could see some use for the many torture devices and experimental surgeries in game.

If you do go with this. Killed the stations pets would be a good one.

**Comment by ArcaneMusic** [comment]
Would you be interested in spawning prisoners in with a short list of contraband items that they smuggled in? Nothing traitorous, but maybe items like stacks of cash, cigarettes, drug seeds, etc.

**Comment by Bawhoppen** [comment]
I think they should spawn with a generated list of crimes they originally committed to get into perma.
This would give the warden another small something to do, needing to look them up on the security computer so they know what they're dealing with. 

Also this may work better if we expanded perma a LITTLE bit on every map. Less of an open design, maybe. 

Otherwise, I think this could be a cool addition! :+1:

**Comment by tralezab** [comment]
i will play the shit out of this role

**Comment by Shadowflame909** [comment]
A job that requires babysitting like this has tons of potential. Maybe when a prisoner wants to argue their crime the courtroom could see more use.

Malf AI's can stealthily cause a prison break and riot thus distracting the crew.

I think the prisoners should try to repeat there crimes like mass station pet murder, destroying Christmas trees and public indecency.

Very nice.

File: code/datums/traits/negative.dm [hunk 3]
```diff
@@ -133,6 +133,8 @@
 				heirloom_type = /obj/item/reagent_containers/food/drinks/bottle/whiskey
 			if("Lawyer")
 				heirloom_type = pick(/obj/item/gavelhammer, /obj/item/book/manual/wiki/security_space_law)
+			if("Prisoner")
+				heirloom_type = /obj/item/pen/blue
```
Comment (Qustinnus) [inline code/datums/traits/negative.dm]: why

**Comment by Ohagi-Chan** [comment]
I've asked admins to spawn me like this on the shuttle a few times, alerting security to go pick up the prisoner in transit.
This makes things like that less repetitive to set up for them, I guess.

File: code/datums/traits/negative.dm (hunk 3 above)
Comment (RaveRadbury) [reply code/datums/traits/negative.dm]: At this point it's a placeholder. Any suggestion for a more appropriate alternative is welcome.

**Comment by HomocidlePancakes** [comment]
Wow, something that made me go “wow” in good way.

**Comment by ATH1909** [comment]
Could we maybe make it so that the perma intercom don't start the round with their microphone wires cut so that prisoners can at least participate in the comms chatter of the round (without using a holopad or something)?

File: code/datums/traits/negative.dm (hunk 3 above)
Comment (Shadowflame909) [reply code/datums/traits/negative.dm]: instant noodles or a carrot shiv would work

**Comment by SkeletalElite** [comment]
Would be cool if this role had more relaxed escalation similar to free golems.

**Comment by cacogen** [comment]
urgh at least make it a ruin not a job

**Comment by evilcyborgmrt** [comment]
Abductors are going to love these guys

**Comment by itseasytosee** [comment]
New traitor objective! Make sure so and so escapes alive and out of custody.

**Comment by wesoda25** [comment]
I think they should be able to roll for antag, but don’t get an uplink. That way security doesn’t have to stress about random prisoners having carp - that’d just create a gross meta routine for each round. I think it’d be lame if they all got pacifist.

Also permas purposefully flawed layout might make this role a bit of a nightmare for security. Prisoners rushing spears to escape every round will definitely be troublesome. Might wanna consider some policy for them and/or flavortext. 

Other than that great job!

**Comment by SkeletalElite** [comment]
> I think they should be able to roll for antag, but don’t get an uplink. That way security doesn’t have to stress about random prisoners having carp - that’d just create a gross meta routine for each round. I think it’d be lame if they all got pacifist.

Either that or make it so they have to get their uplink in some way thats not possible inside perma.

**Comment by necromanceranne** [comment]
If we're getting a prisoner role, this should be the incentive to expand what perma has available to it. Every permabrig feels horribly empty of anything. If it's no longer just a punishment but an opt-in role, that's a great chance to give it some content.

**Comment by IndieanaJones** [comment]
Now THIS is a quality PR.  Great work.

**Comment by TheVekter** [comment]
This gives Warden something to do at roundstart other than sitting with their thumb lodged up their ass. +1

**Comment by TheFakeElon** [comment]
I think it'd be awesome for an admin button that allowed you to force someone to be prisoner at round start with whatever reason said admin chooses

**Comment by nemvar** [comment]
There probably should be an automated annoucement on security radio that there is a prisoner in perma. Otherwise I see security not even noticing it.
Would just be a nice QOL thing since realistically security would aware of this without having to check first.

File: code/datums/traits/negative.dm (hunk 3 above)
Comment (nemvar) [reply code/datums/traits/negative.dm]: soap

File: code/modules/jobs/job_types/prisoner.dm [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/datum/job/prisoner
+	title = "Prisoner"
+	flag = PRISONER
+	department_head = list("Head of security")
+	department_flag = CIVILIAN
+	faction = "Station"
+	total_positions = 3
```
Comment (nemvar) [inline code/modules/jobs/job_types/prisoner.dm]: 3 seems a bit much. 2 seems more reasonable. You want to have at least one free cell that security can use for their own stuff.

**Comment by necromanceranne** [comment]
Okay one big, BIG issue with this is that permabrig on every single map is not radiation storm proof and has absolutely no secure way of avoiding the storms. This was never a problem before, as permabrig has always been equal to death. But given this is a player chosen role, it isn't like you're opting to die every round. Permabrig might finally need rad storm shielding.

**Comment by Shadowflame909** [comment]
The toilet is the only pernabrig place that is r-walled shielded so yeah.

Though good sec would escort the prisoners into the main below permabrig. With the escape pod.

**Comment by necromanceranne** [comment]
That area on meta isn't shielded actually

**Comment by Shadowflame909** [comment]
I promise you it is on some maps.

Codebase fix this!!!

Unless your talking about the toilet. Considering its one wall away from escaping perma into the armory or space. Then lol map design.

**Comment by necromanceranne** [comment]
No I really am talking about the escape pod maint tunnel. Not actually maint.

**Comment by SpinnerMaster** [comment]
If their crimes are already in the system, capital level offences will give the HoS/Warden permission to execute the prisoner. I might add some sort of "Centcom" note/memory/whatever in the HoS/Warden that informs them that they have a prisoner and that Centcom wants them to be alive (not executed by security staff) at the end of the shift.

**Comment by XDTM** [comment]
The roundend report should give a crew greentext if they are properly escorted on the shuttle brig alive.

**Comment by Ty-the-Smonk** [comment]
Ok cool now the real question is how sec escalation will work at all with this, if a prisoner escapes and is wanted and all they have is a spear, are they ok to attack an officer? Or are they expected to surrender?

**Comment by Shadowflame909** [comment]
Tsk tsk tsk. If you can't use them as fodder for a contained xenomorph. What's the point of prisoners.

You choose the role of a prisoner. Escalation has already started simply by you existing. Basically prisoners are gonna be security's ash walkers.

**Comment by Ty-the-Smonk** [comment]
> Tsk tsk tsk. If you can't use them as fodder for a contained xenomorph. What's the point of prisoners.
> 
> You choose the role of a prisoner. Escalation has already started simply by you existing. Basically prisoners are gonna be security's ash walkers.

Inb4 station murderbone

**Comment by XDTM** [comment]
> 
> 
> Tsk tsk tsk. If you can't use them as fodder for a contained xenomorph. What's the point of prisoners.
> 
> You choose the role of a prisoner. Escalation has already started simply by you existing. Basically prisoners are gonna be security's ash walkers.

Would kinda suck to join a round as a prisoner and have the resident paranoid HoS just come in and shoot you to be safe

**Comment by Shadowflame909** [comment]
You are very right. Which is why incentives like a boost in science/cargo points. Or an occasional drop podding of money bags into security,  or even a crew greentext would create paranoia, tension and incentive for risks to make this game work.

**Comment by BadSS13Player** [comment]
> I think they should be able to roll for antag, but don’t get an uplink. That way security doesn’t have to stress about random prisoners having carp - that’d just create a gross meta routine for each round. I think it’d be lame if they all got pacifist.

IMO it sounds way more fun if they have an uplink. Also, it'd be pretty shitty to roll blood brother as a non-prisoner having to work with a prisoner, and it'd also have no way to stealth the "escape alive and not in custody" final objective.
> Also permas purposefully flawed layout might make this role a bit of a nightmare for security. Prisoners rushing spears to escape every round will definitely be troublesome. Might wanna consider some policy for them and/or flavortext.

What's wrong with that? It's Security's fault if they don't do their job properly.

**Comment by trollbreeder** [comment]
In theory, this will make the permabrig a lot nicer and cooler.
In practice, however (bagilites come to mind)...
"Detective me boy! i've visited the permabrig, the prisoner punch stunned me, took my combat shotgun and now i have buckshot schrapnel all over, aargh argh argh argh!"

**Comment by MishaTarkus** [comment]
Would it be possible for the records to start with the Prisoner's Criminal Information? Furthermore, it'd be cool if said crimes were random and you got warned what you got when you start, for that ARR PEE potential. You could be a murderer one shift and a goat shagger the next.

**Comment by RaveRadbury** [comment]
> In practice, however (bagilites come to mind)...
t. player with 1000+ Event Hall connections who's never played on Bagil

**Comment by Cyberboss** [comment]
Good PR.

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Things to be added into this PR:

- [ ] Ensure every perma has rad shielding
- [ ] Generate criminal records for them
- [ ] Prevent them from rolling antag for now (at least those where they would remain a prisoner i.e. traitor)

Other things can wait for the future but are really good ideas:
- Traitor objective
- Contraband
- Right to appeals
- Crew greentext to have them on shuttle brig
- Allow latejoining (Needs to be thought about)
- Option to become a "fugitive" antagonist if escaped (very hard to implement correctly)

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Also there seem to be unrelated icon changes? Though it could just be the conflict

https://github.com/tgstation/tgstation/pull/48819/checks?check_run_id=391905874

**Comment by Tlaltecuhtli** [comment]
can they get antag?

**Comment by Cyberboss** [comment]
It wouldn't make sense for them to have the fugitive antag if they are still in custody. It would if they escaped, but let's leave that for later.

**Comment by BadSS13Player** [comment]
> It wouldn't make sense for them to have the fugitive antag if they are still in custody. It would if they escaped, but let's leave that for later.

Maybe they could have a special prisoner pseudo-antag which can choose to stay alive until the end or escape out of custody.

File: code/modules/jobs/job_types/prisoner.dm [hunk 5]
```diff
@@ -0,0 +1,25 @@
+/datum/job/prisoner
+	title = "Prisoner"
+	flag = PRISONER
+	department_head = list("Head of security")
```
Comment (BadSS13Player) [inline code/modules/jobs/job_types/prisoner.dm]: huh?

**Comment by trollbreeder** [comment]
> > In practice, however (bagilites come to mind)...
> 
> t. player with 1000+ Event Hall connections who's never played on Bagil

I've only heard about bagil mains in rumors, myths, hearsay, singulo and banter, but i heard they were...
[whispering] validhunting powergaming dickwads that are supposedly ruining manuel [/whisper]
which is why i stay away from bagil at all times

File: code/modules/jobs/job_types/prisoner.dm (hunk 5 above)
Comment (trollbreeder) [reply code/modules/jobs/job_types/prisoner.dm]: "You dont have any control over me, security man! You're not the HoS!"
- John Tiderton, shortly before getting his legs amputated by the warden

```suggestion
	department_head = list("the security team")
```

**Comment by itseasytosee** [comment]
I can't +1 enough this has potential for really cool features

**Comment by Michael-Ashfield** [comment]
This could be a ghost role to avoid all the antag issues and people being randomly assigned to be prisoner.

**Comment by GravitronBlue** [comment]
I agree with the prisoner role in concept, as for implementation I'm rather cautious about there being a roundstart role that takes up one or a couple of the limited number of permabrig cells security has available.
Though my argument sounds worthless when nearly all permabrig worthy cases are instead turned into "execute them because '[valid](https://i.imgur.com/OiLZlxg.png)'".

> This could be a ghost role to avoid all the antag issues and people being randomly assigned to be prisoner.

There used to be a prisoner role as part of a crashed prisoner transport on lavaland which was later removed for misc reasons (I think due to just being free midround respawns).
Fugitives for now seem like a proper way for this to work and comes with strings attached for those who join up.
On that note there's a bit of misc backstories if you want to have a [look](https://github.com/tgstation/tgstation/pull/18398/files#diff-5701dfbcb6f3b2b34f2a26fab4a38eb5R210).

One way I've thought of to work around previous issues is to expand the labor camp on lavaland
to accommodate the new role and actually make it difficult to escape from without prior preparation or outside help. (To those who might complain about making escapes more difficult do I really need to explain why [this](https://i.imgur.com/zlLtKQN.png) is bad?)

Right now however this would lead to new issues due to interaction with temporary labor camp prisoners and being able to hitch a ride back to the station with them, or just having stacks of metal waiting to be claimed as points.

Can't say my own suggestion is viable unless someone has solution to its problems.

Regardless this PR at least deserves a testmerge to see how much interest there is and how ~~bad 
 it could really be~~ much FUN can be had, and at the very least it should work well as a punishment role.

**Comment by Farquaar** [comment]
Love this idea. I have fond memories of being locked in perma after a failed antag run. You really have to make do with what you got.

**Comment by Teepeepee** [comment]
the backpack they spawn with should be okay either way (having it or not) since you can grow cotton and wood for the loom in perma and make one anyways

**Comment by 81Denton** [comment]
@RaveRadbury Are you OK with armhulen taking over the PR? He'd deal with the merge conflicts etc

**Comment by cacogen** [comment]
>I have fond memories of being locked in perma after a failed antag run. You really have to make do with what you got.

Doing what? If it was like a locked room puzzle with a different solution every time I'd get it. But from my innumerable experiences over the years you just mill about until you go braindead. Nowadays you can grow stuff first. Nothing in there is going to save you, no matter how much time you spend. You're not getting out without outside intervention.

I've often wished there were more **useful** items prisoners could make, more means of hiding contraband and more contact with the outside world. If not a way out if you've been in there long enough. Some puzzle. And some limits on executions to protect that opportunity from powergamers.

**Comment by XDTM** [comment]
It makes sense that perma isn't escapable in normal conditions, or it would be rather pointless from the point of view of security.
An inbuilt prisoner role will likely drive further development in the area, and will probably result in more activities to do while waiting for an opportunity to escape.

**Comment by Shadowflame909** [comment]
> > I have fond memories of being locked in perma after a failed antag run. You really have to make do with what you got.
> 
> Doing what? If it was like a locked room puzzle with a different solution every time I'd get it. But from my innumerable experiences over the years you just mill about until you go braindead. Nowadays you can grow stuff first. Nothing in there is going to save you, no matter how much time you spend. You're not getting out without outside intervention.
> 
> I've often wished there were more **useful** items prisoners could make, more means of hiding contraband and more contact with the outside world. If not a way out if you've been in there long enough. Some random weak spot exploitable with a random crafted makeshift tool that takes time to penetrate. And some limits on executions to protect that opportunity from powergamers.

Leather overcoat has space protection to allow you to survive longer.

That +  vulgaris should help you survive space travel

**Comment by RaveRadbury** [comment]
@81Denton Sure, that sounds fine

**Comment by tralezab** [comment]
well maintainers look like due to explainable anomalies this pr is ready to merge 👍

**Comment by RaveRadbury** [comment]
Alright. This is passing travis and there are no merge conflicts. Something weird is going on with the pubby map diff but I only added the three landmarks so I don't know what's going on with that.

**Comment by RaveRadbury** [comment]
This should be ready for an initial test merge.

File: code/modules/jobs/job_types/prisoner.dm [hunk 6]
```diff
@@ -1,25 +1,24 @@
...
 	uniform = /obj/item/clothing/under/rank/prisoner
 	shoes = /obj/item/clothing/shoes/sneakers/orange
 	id = /obj/item/card/id/prisoner
-	ears = 0
-	belt = 0
-	back = 0
-	box = 0
+	ears = null
+	belt = null
+	box = null
```
Comment (Cyberboss) [inline code/modules/jobs/job_types/prisoner.dm]: The box should probably be present

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Two things that should be handled before this can be merged.

- The role should be excluded from the list of capable antag roles for now to prevent things from becoming complicated. Note, this won't prevent people who pick prisoner from getting roundstart antag due to how job distribution code works.
- Latejoins should either be disabled or prisoners should spawn in buckle-cuffed on the arrivals shuttle so they can't yakety-sax security from the get-go.

**Comment by Shadowflame909** [comment]
> * Latejoins should either be disabled or prisoners should spawn in buckle-cuffed on the arrivals shuttle so they can't yakety-sax security from the get-go.

An easy one line change for this one is spawning prisoners with shoecuffs at roundstart. Very brutal, torture to move with. Requires someone else to take them off of you.

**Comment by optimumtact** [comment]
just make it roundstart only.

It doesn't make any sense they'd ship prisoners out, but it makes it seem like there was life before the shift start.

**Comment by Fikou** [comment]
> * Latejoins should either be disabled or prisoners should spawn in buckle-cuffed on the arrivals shuttle so they can't yakety-sax security from the get-go.

this is already in place, one file has job list + roundstart slots and overall slots, prisoner has 3 roundstart slots but 0 overall so you cant latejoin

**Comment by optimumtact** [comment]
conflicts

**Comment by optimumtact** [comment]
you are doing this on purpose!

**Comment by RaveRadbury** [comment]
I'm going to clear out all of my conflicts now that I have the rest of it taken care of.

**Comment by RaveRadbury** [comment]
Alright, final cleanup

* donut and kilo need to be looked at
* it's trying to delete goat stuff and the constable hat

**Comment by RaveRadbury** [comment]
A monkey on donut and a pingsky on kilo are causing a problem.

**Comment by RaveRadbury** [comment]
This is ready.

**Comment by NanoCats** [comment]
> Extended Shitcurity fix.
> 
> Wish there was a way to make it whitelist only. Or this will turn into assistants 2.0

I mean, won't they just like, be trapped in perma? This is just expediting the process for Assistants

**Pull request merged**